ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/trielog/AbstractTrieLogManager.java

```diff
@@ -88,6 +90,9 @@ public synchronized void saveTrieLog(
       } finally {
         if (success) {
           stateUpdater.commit();
+          trieLogPruner.rememberTrieLogKeyForPruning(
+              forBlockHeader.getNumber(), forBlockHeader.getBlockHash().toArrayUnsafe());
```

Comment:
since block number isn't unique for trielogs, we will get collisions during reorgs

Reply:
yep, the data structure hold forks for pruning too (it's a multimap)